### ISSUE TITLE
Accessibility: Speech to text

### ISSUE THREAD

**Issue by connorscarolyns** [cds-snc/platform-forms-client#4689]
Users with limited mobility would benefit from the ability to enter text via a microphone instead of typing.

https://app.makeitfable.com/share/670e9b95fa0a0c5fe7e051f7
https://app.makeitfable.com/share/670e9b95fa0a0c5fe7e051f7

**Comment by thiessenp-cds** [comment]
Another interesting and important feature. Thinking about implementation the speech-to-text could be enabled when a user tabs into an input but this could be confusing for users not using speech-to-text. It could also be enabled by a button that becomes visible when a user tabs to an input. But again that could be confusing for some users. 
I wonder if this feature could be enabled in an accessibility settings dialog (see #4688)? 

But probably the biggest hurdle will be the privacy, especially for protected forms. Currently the only way I can think of to implements this would be using a WebAPI (Here is a prototype #2596). Since this API would be provided by the browser, in the case of Chrome, that would be Google and their privacy policy would not meet ours (data would be saved and used by third parties). Perhaps this could be available for non-protected forms along with a clear privacy warning?

**Comment by connorscarolyns** [comment]
A good question for @hegazynourhan can we transfer the risk to the user to make an informed decision on if they leverage their browser's WebAPI. 

Users who require this type of support are likely using similar unsafe products to achieve the same results already, as there is no other option for them. Does the benefit to these users outweigh the risk? 

Would we need to draw the line at Protected A? Could we let users know not to use the text-to-speech option for any sensitive information they wouldn't want a 3rd party to intercept?

**Comment by connorscarolyns** [comment]
Design systems draft accessibility panel:
<img width="720" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBd012Q0E9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--12f41936d22115020b0cef88c2baf5ddd8def56a/image.png" alt="image.png" />

**Comment by anikbrazeau** [comment]
Latest speech-to-text (STT) proof of concept prototype: 

> https://gcdigital.slack.com/archives/C01GFD150MC/p1789746025582729
> https://github.com/cds-snc/platform-forms-client/pull/7827

> The goal of this PoC was to see if STT on a Web app could be done outside of a WebAPI (all end at a Google service = privacy issues). Yes it's possible thanks to AI.
> 
> Or in more detail. The browser records the audio. The audio is sent to a local FastAPI service, where an AI model converts it into text. In the current POC, processing happens locally using the Systran/faster-whisper-small model and CPU execution. No browser Web Speech API or third-party browser speech provider is used.
> 
> Any next steps get this running on Infrastructure would require GPU-backed ECS capacity e.g. GPU-capable instance such as a g4dn ($$$). Then lots of work to harden the service, especially security etc. So stopping here for now. :)
